<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Profilescreen</class>
 <widget class="QWidget" name="Profilescreen">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>633</width>
    <height>471</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Insulin Delivery Profiles</string>
  </property>
  <property name="styleSheet">
   <string notr="true">QLabel {
    font-weight: bold;
}

QPushButton {
    border: 1px solid #555555;
    border-radius: 5px;
    padding: 5px 10px;
}

QPushButton:hover {
    background-color: #555555;
	color: #ffffff;
}

QPushButton:pressed {
    background-color: #777777;
}</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="geometry">
    <rect>
     <x>-10</x>
     <y>30</y>
     <width>649</width>
     <height>431</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="mainLayout">
    <property name="spacing">
     <number>6</number>
    </property>
    <item>
     <widget class="QWidget" name="profileSelectionPanel" native="true">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>280</width>
        <height>429</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>280</width>
        <height>429</height>
       </size>
      </property>
      <layout class="QVBoxLayout" name="leftPanelLayout">
       <item>
        <layout class="QVBoxLayout" name="profileSelectionFrame">
         <item>
          <widget class="QLabel" name="profileHeaderLabel">
           <property name="text">
            <string>Profile Selection Menu</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="instructionsLabel">
           <property name="text">
            <string>Select a profile to view, or use the buttons below to manage profiles.</string>
           </property>
           <property name="wordWrap">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QListWidget" name="profileList">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>200</height>
            </size>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="profileButtonLayout">
           <item>
            <widget class="QPushButton" name="profileAddBtn">
             <property name="font">
              <font>
               <pointsize>8</pointsize>
              </font>
             </property>
             <property name="text">
              <string>Add</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="profileEditBtn">
             <property name="font">
              <font>
               <pointsize>8</pointsize>
              </font>
             </property>
             <property name="text">
              <string>Edit</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="profileDeleteBtn">
             <property name="font">
              <font>
               <pointsize>8</pointsize>
              </font>
             </property>
             <property name="text">
              <string>Delete</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="currentProfileBtn">
             <property name="font">
              <font>
               <pointsize>8</pointsize>
              </font>
             </property>
             <property name="text">
              <string>Select</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QWidget" name="profileEditPanel" native="true">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="visible">
       <bool>true</bool>
      </property>
      <layout class="QVBoxLayout" name="rightPanelLayout">
       <item>
        <layout class="QVBoxLayout" name="profileEditFrame">
         <item>
          <widget class="QLabel" name="profileEditHeaderLabel">
           <property name="text">
            <string>Profile Edit Menu</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QFormLayout" name="profileNameLayout">
           <item row="0" column="0">
            <widget class="QLabel" name="profileNameLabel">
             <property name="text">
              <string>Profile Name:</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QLineEdit" name="profileNameLineEdit">
             <property name="text">
              <string/>
             </property>
             <property name="placeholderText">
              <string>e.g., Morning Routine, Exercise Mode</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QGroupBox" name="groupBolusSettings">
           <property name="title">
            <string>Bolus Settings</string>
           </property>
           <layout class="QFormLayout" name="bolusSettingsLayout">
            <item row="0" column="0">
             <widget class="QLabel" name="carbRatioLabel">
              <property name="text">
               <string>Carb Ratio (g/U):</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QSpinBox" name="carbRatioSpin">
              <property name="maximum">
               <number>1000</number>
              </property>
              <property name="value">
               <number>10</number>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="correctionFactorLabel">
              <property name="text">
               <string>Correction Factor:</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QSpinBox" name="correctionFactorSpin">
              <property name="maximum">
               <number>1000</number>
              </property>
              <property name="value">
               <number>50</number>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="targetGlusoceFactorLabel">
              <property name="text">
               <string>Target Glucose (mg/dL):</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QSpinBox" name="targetGlusoceSpin">
              <property name="maximum">
               <number>1000</number>
              </property>
              <property name="value">
               <number>110</number>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="groupBasalRate">
           <property name="title">
            <string>Basal Rate Settings</string>
           </property>
           <layout class="QFormLayout" name="basalRateLayout">
            <item row="0" column="0">
             <widget class="QLabel" name="basalRateLabel">
              <property name="text">
               <string>Basal Rate (U/hour):</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QDoubleSpinBox" name="basalRateSpin">
              <property name="decimals">
               <number>1</number>
              </property>
              <property name="minimum">
               <double>0.0</double>
              </property>
              <property name="maximum">
               <double>90.0</double>
              </property>
              <property name="singleStep">
               <double>0.1</double>
              </property>
              <property name="value">
               <double>0.5</double>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="buttonLayout">
           <item>
            <widget class="QPushButton" name="cancelBtn">
             <property name="font">
              <font>
               <pointsize>8</pointsize>
              </font>
             </property>
             <property name="text">
              <string>Cancel</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="saveProfileBtn">
             <property name="font">
              <font>
               <pointsize>8</pointsize>
              </font>
             </property>
             <property name="text">
              <string>Save Profile</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>3</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
